monai/transforms/io/array.py

```diff
@@ -478,6 +503,8 @@ def __call__(self, img: torch.Tensor | np.ndarray, meta_data: dict | None = None
                 )
             else:
                 self._data_index += 1
+                if self.savepath_in_metadict and meta_data is not None:
```

Comment:
Hi @wyli ,

I don't remember clearly, why we support multiple writers here with a `for loop`?
If so, I think we should not update "self._data_index += 1" for every writer, is it a bug?

And @MathijsdeBoer , here if we save the "filename" to `meta_data`, will this information be saved into the nifti file header? If yes, I think we should only save the filename instead of the full path which may contain the user's personal information?

Thanks.

Reply:
I was under the impression that saving things to the `meta_data` was fine. A quick skim of the `Writer` code, it seems like the `NiBabelWriter` specifically gets certain keys, rather than taking them all. 
Additionally, I believe that the Nifti specification is quite strict with what the header can contain, i.e.: very little to no room for custom data. If this is also true in the case for, for example, a `.mhd/.mha`, which is a little more freeform, I don't know.
Finally, the key is inserted into the `meta_data` _after_ writing the file to disk, which would not include it in the resulting file either way.